.github/workflows/metadata-ingestion.yml

```diff
@@ -55,6 +55,22 @@ jobs:
           hadoop-version: "3.2"
       - name: Install dependencies
         run: ./metadata-ingestion/scripts/install_deps.sh
+      - name: Calculate pip install plan
+        id: "packages_checksum"
+        run: |
+          cd metadata-ingestion
+          pip install pip -U # only 22.2 and above contains the --dry-run flag
+
+          # only the last line of the output is the packages that will be installed
+          pip install --dry-run -e .  | tail -n 1 > /tmp/would_be_installed.txt
```

Comment:
the pip extra should be included here as well

Reply:
the other thing that I'm thinking about - the pip install dry run still needs to download the packages to inspect their dependency metadata, so it may still be useful to cache the pip cache directory in addition to these changes

Reply:
@hsheth2 I think these are addressed now